src/cmd/healthcheck/main.go

```diff
@@ -21,3 +27,39 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func checkerHTTPClientProvider(config *cmd.Config) *http.Client {
+	if config.ProxyProtocol {
+		return &http.Client{
+			Transport: &http.Transport{
+				DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+					conn, err := net.Dial(network, addr)
+					if err != nil {
+						return nil, err
+					}
+
+					header := proxyproto.Header{
+						Command: proxyproto.LOCAL,
+						Version: 2,
+					}
+					_, err = header.WriteTo(conn)
+					if err != nil {
+						return nil, err
+					}
+					return conn, nil
+				},
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			},
+		}
+	} else {
+		return &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			},
+		}
+	}
```

Comment:
Minor suggestion here -- it would be clearer to build the `http.Client` first, and then just assign to `client.Transport.DialContext` when `config.ProxyProtocol` is `true`.  It just makes it clear that the only difference between the two paths is the `DialContext`.

Reply:
Can't quite do that, but I can create the transport, and then add the `transport.DialContext` when `config.ProxyProtocol` is `true`.  Updated to do that though.